src/ResourceManager/Common/Commands.Common.Strategies/AsyncCmdletExtensions.cs

```diff
@@ -0,0 +1,187 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.Commands.Common.Strategies
+{
+    public static class AsyncCmdletExtensions
+    {
+        public static void WriteVerbose(this IAsyncCmdlet cmdlet, string message, params object[] p)
+            => cmdlet.WriteVerbose(string.Format(message, p));
+
+        public static string UpdateLocation(
+            this IState current, string location, IResourceConfig config)
+            => location ?? current.GetLocation(config) ?? "eastus";
```

Comment:
do we always want to default to eastus for all usages of the strategies library?

Reply:
I'm not sure. We need to discuss this.